### PR TITLE
RHICOMPL-321 - Block import of xccdf.xml reports

### PR DESCRIPTION
**Pull request by dLobatog**
In draft for now, needs testing.

### CODE REVIEW

File: app/services/xccdf_report_parser.rb [hunk 1]
```diff
@@ -25,6 +27,12 @@ def initialize(report_contents, message)
     @metadata = message['metadata']
     @host_inventory_id = message['id']
     @oscap_parser = OpenscapParser::Base.new(report_contents)
+    check_report_format
+  end
+
+  def check_report_format
+    raise WrongFormatError if @oscap_parser.report_xml.
+      at_xpath('.//benchmark/@href').value.match?('.*-xccdf.xml')
```
Comment (akofink) [inline app/services/xccdf_report_parser.rb]: To be safe, we should look only for benchmarks nested under a TestResult. xpath supports matching as well:

```rb
.xpath(".//TestResult/benchmark[contains(@href, '-xccdf.xml')]/@href").any?
```

Comment (dLobatog) [reply app/services/xccdf_report_parser.rb]: I can do that for sure :smile:  there was only 1 match for this xpath as far as I could see but it's a good point.

Comment (akofink) [reply app/services/xccdf_report_parser.rb]: There _should_ only ever be one `<TestResult />` afaik, but I could be wrong.

File: test/services/xccdf_report_parser_test.rb [hunk 2]
```diff
@@ -216,4 +216,21 @@ class XCCDFReportParserTest < ActiveSupport::TestCase
       end
     end
   end
+
+  context 'datastream-based reports only' do
+    should 'raise an error if the report is not coming from a datastream' do
+      fake_report = file_fixture('rhel-xccdf-report-wrong.xml').read
```
Comment (dLobatog) [inline test/services/xccdf_report_parser_test.rb]: This is just a report coming from a xccdf file.

**Pull request merged**